### DISCUSSION TITLE
How to extend request resolution timeout?

### DISCUSSION THREAD

**Discussion by gAlleb** [savonet/liquidsoap#4195] [Q&A] [answered]
When using autocue (cue_file) in 2.3.x an trying to process very long tracks (>1h) I'm getting

```
[request:2] Time limit exceeded by 20.64 secs (timeout: 30.00)!

```

How I can change this timeout? Pretty sure we can extend the request resolution timeout.

P.S. it is a "playlist" operator.

**Comment by toots** [comment, marked as answer]
I've created a PR adding a new `settings.request.timeout` global configuration setting for that: https://github.com/savonet/liquidsoap/pull/4196

I have to say, though, that request resolution should be fast by default. If your request resolution is too long, this will cause all sort of complications.

If the resolution duration is because of some processing such as `autocue`, I would strongly recommend pre-processing your files.

**Reply by gAlleb** [reply]
Thanks! 
I do understand the consequences)

**Reply by gAlleb** [reply]
However, few months ago there was a similar problem in Azuracast with their `request.dynamic` function. While `autocue cue_file` has default timeout of `60.` - `request.dynamic` couldn't cope with long tracks. Had a an idea of adding the same timeout to that function as well. Now it looks like:

```
settings.autocue.cue_file.timeout := 60

dynamic = request.dynamic(id="next_song", timeout=settings.autocue.cue_file.timeout(), retry_delay=10., azuracast.autodj_next_song)
```

and accepts long tracks. Been working fine since.
